lib/web_ui/lib/src/engine/bitmap_canvas.dart

```diff
@@ -299,50 +300,58 @@ class BitmapCanvas extends EngineCanvas {
 
   @override
   void drawLine(ui.Offset p1, ui.Offset p2, SurfacePaintData paint) {
-    _applyPaint(paint);
-    _canvasPool.strokeLine(p1, p2);
+    _withPaint(paint, () {
```

Comment:
will likely allocate a new class instance for each call due to closure.

Reply:
Done. Changed it to `_setUpPaint`/`_tearDownPaint` methods.